src/coap_mbedtls.c

```diff
@@ -1837,6 +1838,41 @@ coap_tls_version_t * coap_get_tls_library_version(void)
   return &version;
 }
 
+coap_digest_ctx_t *
+coap_digest_setup(void) {
+  mbedtls_sha256_context *digest_ctx = mbedtls_malloc(sizeof(mbedtls_sha256_context));
+
+  if (digest_ctx) {
+    mbedtls_sha256_init(digest_ctx);
+    mbedtls_sha256_starts_ret(digest_ctx, 0);
+  }
+  return digest_ctx;
+}
+
+void
+coap_digest_free(coap_digest_ctx_t *digest_ctx) {
+  mbedtls_sha256_free(digest_ctx);
+  mbedtls_free(digest_ctx);
+}
+
+int
+coap_digest_update(coap_digest_ctx_t *digest_ctx,
+                   const uint8_t *data,
+                   size_t data_len) {
+  int ret = mbedtls_sha256_update_ret(digest_ctx, data, data_len);
+
+  return ret == 0 ? 1 : 0;
+}
+
+int
+coap_digest_final(coap_digest_ctx_t *digest_ctx,
+                         coap_digest_t *digest_buffer) {
+  int ret = mbedtls_sha256_finish_ret(digest_ctx, (uint8_t*)digest_buffer);
+
+  coap_digest_free(digest_ctx);
+  return ret == 0 ? 1 : 0;
```

Comment:
`return ret == 0;` seems clearer to me, using the implicit conversion from bool to int.

Reply:
updated.